zenoh/tests/matching.rs

```diff
@@ -26,200 +28,212 @@ macro_rules! ztimeout {
     };
 }
 
+#[cfg(feature = "unstable")]
+async fn create_session_pair(locator: &str) -> Result<(Session, Session)> {
+    let config1 = {
+        let mut config = zenoh::config::peer();
+        config.scouting.multicast.set_enabled(Some(false)).unwrap();
+        config
+            .listen
+            .set_endpoints(vec![locator.clone().parse().unwrap()])
+            .unwrap();
+        config
+    };
+    let config2 = zenoh::config::client([Locator::from_str(locator)?]);
+
+    let session1 = ztimeout!(zenoh::open(config1).res_async())?;
+    let session2 = ztimeout!(zenoh::open(config2).res_async())?;
+    Ok((session1, session2))
+}
+
 #[cfg(feature = "unstable")]
 #[test]
-fn zenoh_matching_status_any() {
+fn zenoh_matching_status_any() -> Result<()> {
     use flume::RecvTimeoutError;
 
     task::block_on(async {
         zasync_executor_init!();
 
-        let session1 = ztimeout!(zenoh::open(config::peer()).res_async()).unwrap();
-
-        let session2 = ztimeout!(zenoh::open(config::peer()).res_async()).unwrap();
+        let (session1, session2) = create_session_pair("tcp/127.0.0.1:18001").await?;
 
         let publisher1 = ztimeout!(session1
             .declare_publisher("zenoh_matching_status_any_test")
             .allowed_destination(Locality::Any)
-            .res_async())
-        .unwrap();
+            .res_async())?;
```

Comment:
In case of tests it's better to use `unwrap()` because the error message will tell exactly at which line the test has panicked. If `?` is used, it's then harder to identify which is the line that fails.

Reply:
Okay. Then I understand why we have our customized zerror since it prints more information than common errors.